src/components/flat-table/sort/sort.component.tsx

```diff
@@ -40,9 +42,13 @@ const Sort = ({ children, onClick, sortType }: SortProps) => {
       >
         {children}
         {sortType && (
-          <Icon
+          <StyledSortIcon
             type={sortType === "ascending" ? "sort_up" : "sort_down"}
-            color="--colorsUtilityMajor200"
+            iconColor={
+              colorTheme === "dark"
+                ? "--colorsActionMinorYang100"
```

Comment:
**issue**: it seems the dark style rule is being overridden by another more specific one:

![Screenshot 2024-03-05 at 17 03 08](https://github.com/Sage/carbon/assets/18368713/d2b347bc-fc93-43b9-8853-eb9bc6b74714)

Looking at the applied CSS rules in the browser console, it appears [this rule within `StyledFlatTableRow`](https://github.com/Sage/carbon/blob/e474046cdf8dec7fea4294b5f7647ed8b3856bdf/src/components/flat-table/flat-table-row/flat-table-row.style.ts#L188-L190) is the culprit:

```css
[data-component="icon"]:not([color]) {
    color: var(--colorsActionMinor500);
}
``` 

One potential way to fix this would be to override the icon colour to be this token within [`StyledCellContent`](StyledCellContent) instead.